markets/storageadapter/ondealsectorcommitted.go

```diff
@@ -137,26 +138,47 @@ func (mgr *SectorCommittedManager) OnDealSectorPreCommitted(ctx context.Context,
 			return true, nil
 		}
 
-		// Extract the message parameters
-		var params miner.SectorPreCommitInfo
-		if err := params.UnmarshalCBOR(bytes.NewReader(msg.Params)); err != nil {
-			return false, xerrors.Errorf("unmarshal pre commit: %w", err)
-		}
-
 		// When there is a reorg, the deal ID may change, so get the
 		// current deal ID from the publish message CID
 		res, err := mgr.dealInfo.GetCurrentDealInfo(ctx, ts.Key().Bytes(), &proposal, publishCid)
 		if err != nil {
 			return false, err
 		}
 
-		// Check through the deal IDs associated with this message
-		for _, did := range params.DealIDs {
-			if did == res.DealID {
-				// Found the deal ID in this message. Callback with the sector ID.
-				cb(params.SectorNumber, false, nil)
-				return false, nil
+		// Extract the message parameters
+		switch msg.Method {
+		case miner.Methods.PreCommitSector:
+			var params miner.SectorPreCommitInfo
```

Comment:
suggestion: this feels like it belongs in a function `onSectorNumber(msg.Methods, msg.Params, res.DealID, cb)`